### PR TITLE
feat: Add target permission when creating a news target - EXO-60503

### PR DESCRIPTION
**Pull request by IlhemEssaadi**

### CODE REVIEW

File: services/src/main/resources/locale/portlet/news/News_en.properties [hunk 1]
```diff
@@ -239,9 +239,14 @@ news.publishTargets.management.addTarget=Add a target
 news.publishTargets.management.editTarget=Edit a target
 news.publishTargets.managementDrawer.name=Name
 news.publishTargets.managementDrawer.description=Description
+news.publishTargets.managementDrawer.permissions=Permissions
+news.publishTargets.managementDrawer.tooltip.permissions=Only publishers of the selected groups and spaces will be able to publish in this news target.
```
Comment (azayati) [inline services/src/main/resources/locale/portlet/news/News_en.properties]:
```suggestion
news.publishTargets.managementDrawer.permissions.tooltip=Only publishers of the selected groups and spaces will be able to publish in this news target.
```

File: services/src/main/resources/locale/portlet/news/News_en.properties [hunk 2]
```diff
@@ -239,9 +239,14 @@ news.publishTargets.management.addTarget=Add a target
...
+news.publishTargets.managementDrawer.explicativeSentence.permissions=Select the publishers of spaces or groups authorized to publish articles in this target
+news.publishTargets.managementDrawer.noDataLabel=No data
```
Comment (azayati) [inline services/src/main/resources/locale/portlet/news/News_en.properties]:
```suggestion
news.publishTargets.managementDrawer.permissions.noData=No data
```

File: services/src/main/resources/locale/portlet/news/News_en.properties [hunk 3]
```diff
@@ -239,9 +239,14 @@ news.publishTargets.management.addTarget=Add a target
 news.publishTargets.management.editTarget=Edit a target
 news.publishTargets.managementDrawer.name=Name
 news.publishTargets.managementDrawer.description=Description
+news.publishTargets.managementDrawer.permissions=Permissions
+news.publishTargets.managementDrawer.tooltip.permissions=Only publishers of the selected groups and spaces will be able to publish in this news target.
+news.publishTargets.managementDrawer.explicativeSentence.permissions=Select the publishers of spaces or groups authorized to publish articles in this target
```
Comment (azayati) [inline services/src/main/resources/locale/portlet/news/News_en.properties]:
```suggestion
news.publishTargets.managementDrawer.permissions.description=Select the publishers of spaces or groups authorized to publish articles in this target
```

File: webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue [hunk 4]
```diff
@@ -105,19 +156,45 @@ export default {
...
+      return (this.selectedTarget.targetName === this.targetName && this.selectedTarget.targetDescription === this.targetDescription) || this.checkAlphanumeric !== '' || this.targetName.length === 0 || this.permissions.length === 0 || this.sameTargetError || this.targetDescription.length > this.targetDescriptionTextLength;
     },
     saveButtonLabel() {
       return this.saveMode === 'edit' ? this.$t('news.publishTargets.managementDrawer.btn.update') : this.$t('news.publishTargets.managementDrawer.btn.confirm');
-    }
+    },
+    isMobile() {
+      return this.$vuetify.breakpoint.name === 'xs' || this.$vuetify.breakpoint.name === 'sm';
+    },
+
```
Comment (azayati) [inline webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue]: remove empty line

File: webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue [hunk 5]
```diff
@@ -162,10 +264,19 @@ export default {
         name: '',
         properties: ''
       };
+      let permissions = '';
+
+      if (this.permissions.length > 0) {
+        this.permissions.forEach(permission => {
+          permissions = `${permissions + permission.remoteId  },`;
+        });      } 
+
```
Comment (azayati) [inline webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue]: remove empty lines

File: webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue [hunk 6]
```diff
@@ -162,10 +264,19 @@ export default {
         name: '',
         properties: ''
       };
+      let permissions = '';
+
+      if (this.permissions.length > 0) {
+        this.permissions.forEach(permission => {
+          permissions = `${permissions + permission.remoteId  },`;
+        });      }
```
Comment (azayati) [inline webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue]: format code

File: webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue [hunk 7]
```diff
@@ -162,10 +264,19 @@ export default {
         name: '',
         properties: ''
       };
+      let permissions = '';
+
```
Comment (azayati) [inline webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue]: remove empty line

File: webapp/src/main/webapp/news-publish-targets-management/initComponents.js [hunk 8]
```diff
@@ -16,10 +16,13 @@
  */
 import NewsPublishTargetsManagement from './components/NewsPublishTargetsManagement.vue';
 import NewsPublishTargetsManagementDrawer from './components/NewsPublishTargetsManagementDrawer.vue';
+import NewsPublishTargetsManagementPermissions from './components/NewsPublishTargetsManagementPermissions.vue';
+
```
Comment (azayati) [inline webapp/src/main/webapp/news-publish-targets-management/initComponents.js:20]: remove empty lines

**Review by azayati — CHANGES_REQUESTED** [review]
see review

File: services/src/main/resources/locale/portlet/news/News_en.properties (hunk 2 above)
Comment (azayati) [reply services/src/main/resources/locale/portlet/news/News_en.properties]: not fixed @IlhemEssaadi

**Review by azayati — CHANGES_REQUESTED** [review]
https://github.com/exoplatform/news/pull/686#discussion_r1037225736

File: services/src/main/resources/locale/portlet/news/News_en.properties (hunk 2 above)
Comment (IlhemEssaadi) [reply services/src/main/resources/locale/portlet/news/News_en.properties]: > not fixed @IlhemEssaadi

Done !!

File: webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue [hunk 9]
```diff
@@ -105,19 +157,47 @@ export default {
...
+    permissions: [],
   }),
   computed: {
+    ignoredItems() {
+      return this.permissions.map(permission => `${permission.providerId}:${permission.remoteId}`);
+    },
+    suggesterLabels() {
+      return {
+        placeholder: this.$t('news.publishTargets.managementDrawer.placeholder.permissions'),
+        noDataLabel: this.$t('news.publishTargets.managementDrawer.permissions.noDataLabel')
```
Comment (azayati) [inline webapp/src/main/webapp/news-publish-targets-management/components/NewsPublishTargetsManagementDrawer.vue]:
```suggestion
        noDataLabel: this.$t('news.publishTargets.managementDrawer.permissions.noData')
```

**Review by azayati — CHANGES_REQUESTED** [review]

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**